### PR TITLE
Feature/meta operations errors

### LINKED ISSUE #67: Better handling errors in conditional meta-operations
When trying to call a meta-operation that is only defined conditionally and the condition is not met, catt just tries to apply it, which produces an invalid term. This results in cryptic errors. A better way would be to check beforehand that the condition for the meta-operation are met, and report a adequate error message when it's not the case.

- [x] inverses
- [x] witnesses
- [x] functorialisation
- [x] naturality

### PR DESCRIPTION
**Pull request by thibautbenjamin**
closes #67

### CODE REVIEW

File: lib/functorialisation.ml [hunk 1]
```diff
@@ -17,6 +19,28 @@ module Memo = struct
       res
 end
 
+let check_closed ctx l =
```
Comment (regular-citizen) [inline lib/functorialisation.ml:22]: Can this be called "upper-closed"? Or at least add a comment?

File: lib/functorialisation.ml [hunk 2]
```diff
@@ -246,27 +273,26 @@ and tm c t s =
...
+        "higher-dimensional transformations in depth >= 0 are not yet supported"
+
 (* Functorialisation of a coherence: exposed function *)
-let coh c l =
-  try coh c l
-  with FunctorialiseMeta ->
-    Error.functorialisation
-      ("coherence: " ^ Coh.to_string c)
-      (Printf.sprintf "cannot functorialise meta-variables")
+let coh c l = report_errors (fun _ -> coh c l) ("coherence: " ^ Coh.to_string c)
```
Comment (regular-citizen) [inline lib/functorialisation.ml:289]: Perf concern: building a bunch of strings every time we want to functorialise

**Review by regular-citizen — CHANGES_REQUESTED** [review]
Left some edits I think are important. One thing I think we risk losing here is the ability to exist with error if a meta-operation crashes because it's unsupported. Don't want to add a ton of flags but a file full of things that don't compute should not exit with success

**Comment by thibautbenjamin** [comment]
> Left some edits I think are important. One thing I think we risk losing here is the ability to exist with error if a meta-operation crashes because it's unsupported. Don't want to add a ton of flags but a file full of things that don't compute should not exit with success

It's not what's happening. It exits with failure if the operation is not supported. It only exits with a success in the test file because it is launched with the `--keep-going` flag, which is only for testing purposes

File: lib/functorialisation.ml (hunk 1 above)
Comment (thibautbenjamin) [reply lib/functorialisation.ml:22]: sure, I'll change the name. Not a big fan of adding a comment, the name should be clear enough to be self-explanatory

File: lib/functorialisation.ml (hunk 2 above)
Comment (thibautbenjamin) [reply lib/functorialisation.ml:289]: That's a very good point... maybe this refactoring is not such a good idea, or maybe I can change it to use the `lazy` trick... I'll try to fiddle with it

**Review by regular-citizen — APPROVED** [review]
Approved

**Pull request merged**